lms/djangoapps/learner_dashboard/programs.py

```diff
@@ -242,3 +159,83 @@ def standalone_page_title(self, request, fragment, **kwargs):
         return _('Program Details')
+
+
+class ProgramDiscussionLTI:
+    """
+      Render the program discussion iframe.
+    """
+    DEFAULT_ROLE = 'Student'
+    ADMIN_ROLE = 'Administrator'
+
+    def __init__(self, program_uuid, request):
+        self.program_uuid = program_uuid
+        self.request = request
+        self.enabled = program_discussions_is_enabled()
+
+    def get_program_discussion_configuration(self):
+        return ProgramDiscussionsConfiguration.objects.filter(
+            program_uuid=self.program_uuid
+        ).first()
+
+    def _get_resource_link_id(self) -> str:
+        site = get_current_site(self.request)
+        return f'{site.domain}-{self.program_uuid}'
+
+    def _get_result_sourcedid(self, resource_link_id) -> str:
+        return f'{self.program_uuid}:{resource_link_id}:{self.request.user.id}'
+
+    def get_user_role(self):
+        """
+        Returns the given user's role
+        """
+        if GlobalStaff().has_user(self.request.user):
+            return self.ADMIN_ROLE
+        return self.DEFAULT_ROLE
+
+    def _get_lti_embed_code(self, program_discussions_configuration) -> str:
+        """
+        Returns the LTI embed code for embedding in the program discussions tab
+        Args:
+            program_discussions_configuration (ProgramDiscussionsConfiguration): ProgramDiscussionsConfiguration object.
+        Returns:
+            HTML code to embed LTI in program page.
+        """
+        resource_link_id = self._get_resource_link_id()
+        result_sourcedid = self._get_result_sourcedid(resource_link_id)
+
+        return lti_embed(
+            html_element_id='lti-tab-launcher',
+            lti_consumer=program_discussions_configuration.lti_configuration.get_lti_consumer(),
+            resource_link_id=resource_link_id,
+            user_id=str(self.request.user.id),
+            roles=self.get_user_role(),
+            context_id=self.program_uuid,
+            context_title=self.program_uuid,
+            context_label=self.program_uuid,
+            result_sourcedid=result_sourcedid
+        )
+
+    def render_discussions_iframe(self) -> str:
+        """
+        Returns the program discussion fragment if program discussions configuration exists for a program uuid
+        """
+        program_discussions_configuration = self.get_program_discussion_configuration()
+        if not self.enabled or not program_discussions_configuration:
+            return ''
+        lti_embed_html = self._get_lti_embed_code(program_discussions_configuration)
+        fragment = Fragment(
+            HTML(
+                """
+                <iframe
+                    id='lti-tab-embed'
+                    style='width: 100%; min-height: 800px; border: none'
+                    srcdoc='{srcdoc}'
+                 >
+                </iframe>
+                """
+            ).format(
+                srcdoc=lti_embed_html
+            )
+        )
+        return fragment.content
```

Comment:
Does this work?
The template expects data in the following format
```
return 
{
    'iframe': fragment.content,
    'enabled': True
}
```

Reply:
very valid point.